sdk/component/component.go

```diff
@@ -0,0 +1,71 @@
+package component
+
+import (
+	"context"
+
+	ocsf "github.com/ocurity/dracon/sdk/component/gen/com/github/ocsf/ocsf_schema/v1"
+)
+
+// Helpers interfaces for common functionalities.
+type (
+	// Validator allows validating findings by a specified criteria.
+	Validator interface {
+		// Validate validates the supplied findings and returns an error if invalid.
+		Validate(ctx context.Context, findings []*ocsf.VulnerabilityFinding) error
+	}
+
+	// Reader allows reading findings from a source.
+	Reader interface {
+		// Read reads and parses findings from a source.
+		Read(ctx context.Context) ([]*ocsf.VulnerabilityFinding, error)
+	}
+
+	// Filterer allows filtering out findings by a specified criteria.
+	Filterer interface {
+		// Filter returns filtered findings from the supplied ones.
+		Filter(ctx context.Context, findings []*ocsf.VulnerabilityFinding) ([]*ocsf.VulnerabilityFinding, error)
+	}
+)
+
+// Components interfaces.
+type (
+	// Sourcer defines the behaviour of source components.
+	Sourcer interface {
+		// Source runs an arbitrary step. Useful for interacting with third-party API - i.e. cloning a repository.
+		Source(ctx context.Context) error
+	}
+
+	// Producer defines the behaviour of producer components.
+	// Read -> Process -> Validate -> Store.
+	Producer interface {
+		Reader
+		Validator
```

Comment:
❓ I think it makes sense to only force people to validate on write as otherwise we'd have to have steps for validating inputs and outputs on `Producer`, `Consumer` and `Enricher`. I think that for now this is enough.

Reply:
As discussed, I think we should validate only the input data of the producers, to check if we're feeding them meaningful values. The output shouldn't be validated, because we can't maintain what 3rd party tools produce.

Reply:
output?
Isn't a producers output Smithy-OCSF?

Reply:
@dlicheva that's fair but how do we defend ourselves from having flows where we could write malformed data in the database?

@northdpole yes - but I think it would be better for data integrity to validate what we're about to output in the data destination - thoughts?